model/jpa/src/main/java/org/keycloak/events/jpa/JpaEventStoreProvider.java

```diff
@@ -276,17 +276,21 @@ protected void clearExpiredAdminEvents() {
         CriteriaBuilder cb = em.getCriteriaBuilder();
         CriteriaQuery<RealmAttributeEntity> cr = cb.createQuery(RealmAttributeEntity.class);
         Root<RealmAttributeEntity> root = cr.from(RealmAttributeEntity.class);
-        cr.select(root).where(cb.and(cb.equal(root.get("name"),RealmAttributes.ADMIN_EVENTS_EXPIRATION),cb.greaterThan(root.get("value").as(Long.class),Long.valueOf(0))));
-        Map<Long, List<RealmAttributeEntity>> realms = em.createQuery(cr).getResultStream().collect(Collectors.groupingBy(attribute -> Long.valueOf(attribute.getValue())));
+        // unable to cast the CLOB to a BIGINT in the select for H2 2.x, therefore comparing strings only in the DB, and filtering again in the next statement
+        cr.select(root).where(cb.and(cb.equal(root.get("name"),RealmAttributes.ADMIN_EVENTS_EXPIRATION),cb.notEqual(root.get("value"), "0")));
+        Map<Long, List<RealmAttributeEntity>> realms = em.createQuery(cr).getResultStream()
+                // filtering again on the attribute as paring the CLOB to BIGINT didn't work in H2 2.x
+                .filter(attribute -> Long.parseLong(attribute.getValue()) > 0)
```

Comment:
Thanks for pointing out the possibility for an exception. I updated the code to catch the exception and log a warning with helpful information so an admin could fix this. Not logging anything would lead to not deleted events, which would be a surprise to the admin and they wouldn't be able to locate the cause.

Reply:
BTW: The changes with regards to `VALUE` are specific to H2, so I didn't run pipelines for any other database. Would you think otherwise?

Reply:
I think it still makes sense to run those, especially due to the changes in `JpaEventStoreProvider` class.

Reply:
Otherwise, consider this comment resolved. Thanks for updating the code!